src/gui/sdl_mapper.cpp

```diff
@@ -102,17 +102,14 @@ static void SetActiveEvent(CEvent * event);
 static void SetActiveBind(CBind * _bind);
 extern uint8_t int10_font_14[256 * 14];
 
-static std::vector<CEvent *> events;
-static std::vector<CButton *> buttons;
+static std::vector<std::unique_ptr<CEvent>> events;
+static std::vector<std::unique_ptr<CButton>> buttons;
```

Comment:
It's much appreciated that you made the mapper safer 🎖️

Just a heads up, we're gonna blow away the whole mapper in the future and rebuild it from scratch because frankly, this legacy implementation is rather terrible. I'm just mentioning this so you don't spend too much effort on cleaning it up, but small clean-ups and robustness improvements are always welcome until we rewrite it.

See https://github.com/dosbox-staging/dosbox-staging/issues/3075 and https://github.com/dosbox-staging/dosbox-staging/issues/3075#issuecomment-1786310890 for further info.